### PR TITLE
Feat/pb 92762 self serve phase 4

### PR DESCRIPTION
**Pull request by devirte1**

### CODE REVIEW

File: sdk/src/main/java/com/silanis/esl/sdk/examples/AccountEmailReminderSettingsExample.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        eslClient.getAccountConfigService().deleteAccountEmailReminderSettings();
+        deletedAccountEmailReminderSettings = eslClient.getAccountConfigService().getAccountEmailReminderSettings();
+
+        accountEmailReminderSettings = AccountEmailReminderSettingsBuilder.newEmailReminderSettings()
+                .withIntervalInDays(30)
+                .withRepetitionsCount(2)
+                .withStartInDaysDelay(5)
+                .build();
+
+        //Save account email reminder settings
```
[the remark this answers is not in the document]
Comment (devirte1) [reply sdk/src/main/java/com/silanis/esl/sdk/examples/AccountEmailReminderSettingsExample.java:40]: 1st time we save - to test if save/update works fine. Then we delete and save to make sure delete works fine and next save/update happens without any error

File: sdk/src/main/java/com/silanis/esl/sdk/builder/AccountDesignerSettingsBuilder.java [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+     * Set defaultSignatureType in AccountDesignerSettings.
+     * @param signatureType
+     * <p>
+     * DEFAULT: sign
+     * ALLOWED VALUES : mobile, capture
+     * <p>
+     *
+     * @return This
+     */
+    public AccountDesignerSettingsBuilder withDefaultSignatureType(String signatureType) {
```
[the remark this answers is not in the document]
Comment (devirte1) [reply sdk/src/main/java/com/silanis/esl/sdk/builder/AccountDesignerSettingsBuilder.java:355]: Its a drop down. It has 3 values . 
1 - the default one, "Click to Sign" 
2- "Mobile Signature"
3- "Capture Signature"

Based on the requirements, we pass either of them as a parameter.

Why do we need without method ? and what should be an expected behaviour for this method ?

File: sdk/src/main/java/com/silanis/esl/api/model/AccountEmailReminderSettings.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        implements java.io.Serializable {
+
+    @JsonIgnore
+    public static final String FIELD_START_IN_DAYS_DELAY = "startInDaysDelay";
+    @JsonIgnore
+    public static final String FIELD_INTERVAL_IN_DAYS = "intervalInDays";
+    @JsonIgnore
+    public static final String FIELD_REPETITIONS_COUNT = "repetitionsCount";
+
+    protected Integer _startInDaysDelay ;
```
[the remark this answers is not in the document]
Comment (devirte1) [reply sdk/src/main/java/com/silanis/esl/api/model/AccountEmailReminderSettings.java:18]: I followed the convention from previous implementation of other self serve modules. I believe its an inentional convention.

File: sdk/src/main/java/com/silanis/esl/api/model/AccountEmailReminderSettings.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    protected Integer _intervalInDays ;
+    protected Integer _repetitionsCount ;
+
+    public AccountEmailReminderSettings() {
```
[the remark this answers is not in the document]
Comment (devirte1) [reply sdk/src/main/java/com/silanis/esl/api/model/AccountEmailReminderSettings.java:22]: It is not a method. It's the constructor and it is allowed to be empty and we are required to have it for the object that is getting created during run time. I can add a comment as empty constructor

File: sdk/src/main/java/com/silanis/esl/api/model/AccountDesignerSettings.java [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+
+    public AccountDesignerSettings setDone(Boolean value ){
+        SchemaSanitizer.throwOnNull(FIELD_DONE, value);
+        this._done = value;
+        setDirty(FIELD_DONE);
+        return this;
+    }
+
+    @JsonIgnore
+    public AccountDesignerSettings safeSetDone(Boolean value ){
```
[the remark this answers is not in the document]
Comment (devirte1) [reply sdk/src/main/java/com/silanis/esl/api/model/AccountDesignerSettings.java:91]: It is intentional

File: sdk/src/main/java/com/silanis/esl/sdk/builder/AccountEmailReminderSettingsBuilder.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+import com.silanis.esl.sdk.AccountEmailReminderSettings;
+
+public class AccountEmailReminderSettingsBuilder {
+
+    protected Integer startInDaysDelay = null;
+    protected Integer intervalInDays = null;
+    protected Integer repetitionsCount = null;
+
+    public AccountEmailReminderSettingsBuilder() {
```
[the remark this answers is not in the document]
Comment (devirte1) [reply sdk/src/main/java/com/silanis/esl/sdk/builder/AccountEmailReminderSettingsBuilder.java:11]: Again, it is the same - An Empty constructor. I can add a comment for it

**Review by houjianyu — APPROVED** [review]

**Pull request merged**